distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1607,145 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
                 .resolve(CSW_SOURCE_ID + "-" + metacardId + ".ser")), is(false));
     }
 
+    @Test
+    public void testProductDownloadWithTwoUsers() throws Exception {
+        getSecurityPolicy().configureWebContextPolicy(null,
+                "/=SAML|basic,/solr=SAML|PKI|basic",
+                null,
+                null);
+
+        adminCometDClient = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+        localhostCometDClient = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+
+        String filename = "product4.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(0), 0));
+
+        String filename2 = "product5.txt";
+        String metacardId2 = generateUniqueMetacardId();
+        String resourceData2 = getResourceData(metacardId2);
+
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId2))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId2).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId2))
+                .then(getCswRetrievalHeaders(filename2),
+                        chunkedContentWithHeaders(resourceData2, Duration.ofMillis(0), 0));
```

Comment:
`ChunkedContent.chunkedContentWithHeaders()` was updated in a PR that was merged in yesterday to use a builder pattern. Since you aren't using a delay or failures, it should be a simple change to `ChunkedContentBuilder(resourceData2).build()` once you rebase.